apps/webapp/app/v3/services/createCheckpoint.server.ts

```diff
@@ -140,10 +151,72 @@ export class CreateCheckpointService extends BaseService {
           dependencyFriendlyRunId: reason.friendlyId,
         });
 
-        keepRunAlive = await this.#isRunCompleted(reason.friendlyId);
-
-        if (!keepRunAlive) {
-          await marqs?.acknowledgeMessage(attempt.taskRunId);
+        if (checkpointEvent) {
+          const dependency = await this._prisma.taskRunDependency.findFirst({
+            select: {
+              id: true,
+              taskRunId: true,
+            },
+            where: {
+              taskRun: {
+                friendlyId: reason.friendlyId,
+              },
+            },
+          });
+
+          logger.log("CreateCheckpointService: Created checkpoint WAIT_FOR_TASK", {
+            checkpointId: checkpoint.id,
+            runFriendlyId: reason.friendlyId,
+            dependencyId: dependency?.id,
+          });
+
+          if (!dependency) {
+            logger.error("CreateCheckpointService: Dependency not found", {
+              friendlyId: reason.friendlyId,
+            });
+
+            return {
+              success: true,
+              checkpoint,
+              event: checkpointEvent,
+              keepRunAlive: false,
+            };
+          }
+
+          const childAttempt = await this._prisma.taskRunAttempt.findFirst({
```

Comment:
First check that the TaskRun is finished, get the most recent finished status attempt